api/src/services/security-service.ts

```diff
@@ -76,4 +84,86 @@ export class SecurityService extends DBService {
       )
     );
   }
+
+  /**
+   * Returns the signed URL of a document for which a user has permissions.
+   *
+   * Rules:
+   * non-admin user cannot access the document when:
+   ** - it is pending review, OR
+   ** - user hasn't been granted an exception to every security rule
+   *
+   * non-admin user can access the document when:
+   * - document is not secured
+   * - user has the correct exceptions
+   *
+   *
+   * @param {number} artifactId
+   * @return {*}  {Promise<any>}
+   * @memberof SecurityService
+   */
+  async getSecuredArtifactBasedOnRulesAndPermissions(artifactId: number): Promise<any> {
+    const isSystemUserAdmin = await this.userService.isSystemUserAdmin();
+
+    const userId = this.connection.systemUserId();
+
+    const isDocumentPendingReview = (await this.artifactService.getArtifactById(artifactId)).security_review_timestamp
+      ? false
+      : true;
+
+    if (!isSystemUserAdmin && isDocumentPendingReview) {
+      throw new HTTP403('Documents that are pending review can only be downloaded by administrators.');
```

Comment:
nit: not really a technical thing, and probably more of a Jeremy question, but we may have to review the language on security related errors since we may or may now want to reveal why the secured item is unavailable, since that itself reveals something about its security.

Reply:
Talked to Jeremy.  Changing the tex to 'You do not have access to this document'.